provider/oauth2/tests.py

```diff
@@ -677,14 +678,47 @@ def test_basic_client_backend(self):
         self.assertEqual(BasicClientBackend().authenticate(request).id,
                          2, "Didn't return the right client.")
 
+    def test_basic_client_backend_fail(self):
+        request = type('Request', (object,), {'META': {}})()
```

Comment:
Use `[RequestFactory](https://docs.djangoproject.com/en/1.11/topics/testing/advanced/#the-request-factory)`.

```python
request = RequestFactory().get('')
```